src/media/local-stream.spec.ts

```diff
@@ -0,0 +1,108 @@
+import { BaseEffect } from '@webex/web-media-effects';
+import MediaStreamStub from '../mocks/media-stream-stub';
+import MediaStreamTrackStub from '../mocks/media-stream-track-stub';
+import { mocked } from '../mocks/mock';
+import { createMockedStream } from '../util/test-utils';
+import { LocalStream } from './local-stream';
+
+/**
+ * Gain access to private variables using typescript's `any`.
+ *
+ * @param item - The item to gain access to.
+ * @returns The value entered.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const privateAccess = (item: any): any => item;
+
+/**
+ * Create a mocked track effect.
+ *
+ * @returns A mocked TrackEvent.
+ */
+const createMockedTrackEffect = () => {
+  const effectTrack = mocked(new MediaStreamTrackStub());
+  const effect = {
+    dispose: jest.fn().mockResolvedValue(undefined),
+    load: jest.fn().mockResolvedValue(effectTrack),
+    on: jest.fn(),
+  };
+
+  return { effectTrack, effect };
+};
+
+/**
+ * A dummy LocalStream implementation so we can instantiate it for testing.
+ */
+class TestLocalStream extends LocalStream {}
+
+describe('LocalStream', () => {
+  const mockStream = createMockedStream();
+  let localStream: LocalStream;
+  beforeEach(() => {
+    localStream = new TestLocalStream(mockStream);
+  });
+
+  it('should change the underlying track state based on being muted & unmuted', () => {
+    expect.assertions(2);
+    // Simulate the default state of the track's enabled state.
+    mockStream.getTracks()[0].enabled = true;
+
+    localStream.setMuted(true);
+    expect(mockStream.getTracks()[0].enabled).toBe(false);
+    localStream.setMuted(false);
+    expect(mockStream.getTracks()[0].enabled).toBe(true);
+  });
+});
+
+describe('LocalTrack addEffect', () => {
+  // eslint-disable-next-line jsdoc/require-jsdoc
+  const setup = () => {
+    const mockedStream = mocked(new MediaStreamStub());
+    const localStream = new TestLocalStream(mockedStream as unknown as MediaStream);
+    const { effectTrack, effect } = createMockedTrackEffect();
+    return { mockedStream, localStream, effectTrack, effect };
+  };
+
+  it('loads and uses the effect when there is no loading effect', async () => {
+    expect.hasAssertions();
+
+    const { localStream, effectTrack, effect } = setup();
+
+    const addEffectPromise = localStream.addEffect('test-effect', effect as unknown as BaseEffect);
+
+    await expect(addEffectPromise).resolves.toBeUndefined();
+    expect(privateAccess(localStream).outputStream.getTracks()[0]).toBe(effectTrack);
```

Comment:
I think these uses of privateAccess can be changed too?

Reply:
In these cases I think we want to make sure the output stream has the track that the effect is outputting

Reply:
ok...how about just having `TestLocalStream` expose it?

Reply:
Though, to be honest, I can't remember why `Stream` doesn't expose a getter for `outputStream`, I'm pretty sure we'll end up with one (even it's just an abstract getter at the `Stream` level)

Reply:
I ended up having `TestLocalStream` expose it, I can go back and update if/when the getter is added.  Is that something we want to add on this PR?

Reply:
Ok cool. Nah we'll do that later